src/commands/base/BaseCommand.ts

```diff
@@ -0,0 +1,11 @@
+import { Message } from 'discord.js';
+
+export default abstract class BaseCommand {
+  protected message: Message;
+
+  constructor(message: Message) {
```

Comment:
consider this:
```typescript
interface ICommand {
    run(Message message): void; // Or return combination of response object and optional error
}
```
Then message handler can take a map from string to command instance.
Then it does:
`this.commands[message.text.split(' ')[0]].run(message)`
^ pseudocode

Note that in this case the `MessageHandler` could actually implement `ICommand` itself, in which case I would rename `ICommand` to `IMessageHandler` and `MessageHandler` to `MessageDelegator` and all the commands such as `AddSoundCommand` to `AddMessageUserHandler`.

Let me know what you think.